clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemServerTest.java

```diff
@@ -0,0 +1,1375 @@
+                    checksum(STATUS_CHECKSUM).
+                    mtime(STATUS_MTIME).
+                    sizeBytes(STATUS_FILE_SIZE);
+        }
+        expectListing("repo", "main",
+                      ImmutablePagination.builder().prefix("status/").build(),
+                      objects);
+        expectStatObjectNotFound("repo", "main", "status");
+
+        Path dir = new Path("lakefs://repo/main/status");
+        FileStatus[] fileStatuses = fs.listStatus(dir);
+
+        FileStatus[] expectedFileStatuses = new LocatedFileStatus[totalObjectsCount];
+        for (int i = 0; i < totalObjectsCount; i++) {
+            Path p = new Path(dir + "/file" + i);
+            LakeFSFileStatus fileStatus = new LakeFSFileStatus.Builder(p)
+                    .length(STATUS_FILE_SIZE)
+                    .checksum(STATUS_CHECKSUM)
+                    .mTime(STATUS_MTIME)
+                    .blockSize(Constants.DEFAULT_BLOCK_SIZE)
+                    .physicalAddress(s3Url("/repo-base/status" + i))
+                    .build();
+            expectedFileStatuses[i] = new LocatedFileStatus(fileStatus, null);
+        }
+        Assert.assertArrayEquals(expectedFileStatuses, fileStatuses);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testAppend() throws IOException {
+        fs.append(null, 0, null);
+    }
+
+    /**
+     * rename(src.txt, non-existing-dst) -> non-existing/new - unsupported, should fail with false.  (Test was buggy, FIX!)
+     */
+    @Test
+    public void testRename_existingFileToNonExistingDst() throws IOException, ApiException {
+        Path src = new Path("lakefs://repo/main/existing.src");
+
+        ObjectStats stats = new ObjectStats()
+            .path("existing.src")
+            .sizeBytes(STATUS_FILE_SIZE)
+            .mtime(STATUS_MTIME)
+            .pathType(PathTypeEnum.OBJECT)
+            .physicalAddress(s3Url("existing.src"))
+            .checksum(STATUS_CHECKSUM);
+
+        expectStatObject("repo", "main", "existing.src", stats);
+
+        Path dst = new Path("lakefs://repo/main/non-existing/new");
+
+        expectListing("repo", "main",
+                      ImmutablePagination.builder().prefix("non-existing/").build());
+        expectStatObjectNotFound("repo", "main", "non-existing/new");
+        expectStatObjectNotFound("repo", "main", "non-existing/new/");
+        expectListing("repo", "main",
+                      ImmutablePagination.builder().prefix("non-existing/new/").build());
+        expectStatObjectNotFound("repo", "main", "non-existing");
+        expectStatObjectNotFound("repo", "main", "non-existing/");
+
+        boolean renamed = fs.rename(src, dst);
+        Assert.assertFalse(renamed);
+    }
+
+    @Test
+    public void testRename_existingFileToExistingFileName() throws IOException {
+        String contents = "The quick brown fox jumps over the lazy dog.";
+        byte[] contentsBytes = contents.getBytes();
```

Comment:
Used?

Reply:
Yikes.  It's not!  It wasn't before, but it still isn't used.